src/headers.ts

```diff
@@ -80,12 +85,18 @@ export function withHeaders(
 ): Response {
   const response = new Response(inputResponse.body);
```

Comment:
I believe this will also need an `encodeBody: 'manual'`?

Reply:
Yep, it would get lost otherwise. I refactored this whole function to make it easier to follow (I hope) and add this setting to the `ResponseInit` (33802a6)